src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_TicketingStack/Ticketing_TicketTabNavStack/TicketTabNav_PurchaseTabScreen/TicketTabNav_PurchaseTabScreen.tsx

```diff
@@ -33,32 +35,76 @@ export const TicketTabNav_PurchaseTabScreen = ({navigation}: Props) => {
   const showTicketAssistant = useTicketingAssistantEnabled();
   const analytics = useAnalytics();
 
+  const {remoteTokens} = useMobileTokenContextState();
+  const inspectableToken = findInspectable(remoteTokens);
+  const isMobileTokenEnabled = isMobileToken(inspectableToken);
+
   if (must_upgrade_ticketing) return <UpgradeSplash />;
 
   const onProductSelect = (fareProductTypeConfig: FareProductTypeConfig) => {
     analytics.logEvent('Ticketing', 'Fare product selected', {
       type: fareProductTypeConfig.type,
     });
-    if (
-      fareProductTypeConfig.configuration.requiresLogin &&
-      authenticationType !== 'phone'
-    ) {
-      navigation.navigate('Root_LoginRequiredForFareProductScreen', {
-        fareProductTypeConfig,
-        afterLogin: {
-          screen: 'Root_PurchaseOverviewScreen',
-          params: {
-            fareProductTypeConfig,
-            mode: 'Ticket',
+
+    if (authenticationType !== 'phone') {
+      if (
+        fareProductTypeConfig.configuration.requiresLogin &&
+        fareProductTypeConfig.configuration.requiresTokenOnMobile &&
+        !isMobileTokenEnabled
+      ) {
+        return navigation.navigate('Root_LoginRequiredForFareProductScreen', {
+          fareProductTypeConfig,
+          afterLogin: {
+            screen: 'Root_ActiveTokenOnPhoneRequiredForFareProductScreen',
+            params: {
+              nextScreen: {
+                screen: 'Root_PurchaseOverviewScreen',
+                params: {
+                  fareProductTypeConfig,
+                  mode: 'Ticket',
+                },
+              },
+            },
           },
-        },
-      });
+        });
+      }
+
+      if (fareProductTypeConfig.configuration.requiresLogin) {
+        return navigation.navigate('Root_LoginRequiredForFareProductScreen', {
+          fareProductTypeConfig,
+          afterLogin: {
+            screen: 'Root_PurchaseOverviewScreen',
+            params: {
+              fareProductTypeConfig,
+              mode: 'Ticket',
+            },
+          },
+        });
+      }
     } else {
-      navigation.navigate('Root_PurchaseOverviewScreen', {
-        fareProductTypeConfig: fareProductTypeConfig,
-        mode: 'Ticket',
-      });
+      if (
+        fareProductTypeConfig.configuration.requiresTokenOnMobile &&
+        !isMobileTokenEnabled
+      ) {
+        return navigation.navigate(
+          'Root_ActiveTokenOnPhoneRequiredForFareProductScreen',
+          {
+            nextScreen: {
+              screen: 'Root_PurchaseOverviewScreen',
+              params: {
+                fareProductTypeConfig,
+                mode: 'Ticket',
+              },
+            },
+          },
+        );
+      }
     }
+
+    return navigation.navigate('Root_PurchaseOverviewScreen', {
```

Comment:
Shouldn't have `return` on the navigation events. It makes it seem that the `navigate` method has a return value.

Reply:
Fixed